client/src/components/QandA/QandAContainer.jsx

```diff
@@ -42,11 +42,13 @@ function QandA({ product }) {
     }
   };
 
-  useLayoutEffect(() => {
-    getProduct();
+  useEffect(() => {
+    if (product.id) {
+      getProduct();
+    }
   }, [product.id]);
 
-  useLayoutEffect(() => {
```

Comment:
Thanks for changing this back to useEffect. I don't think anyone else was using useLayoutEffect so best to stay consistent with the rest of the repo.